### PR TITLE
add find_predecessor_node_by_edge.

### PR DESCRIPTION
**Pull request by ewinston**
add function to get immediate predecessor of node by edge.

### CODE REVIEW

File: src/digraph.rs [hunk 1]
```diff
@@ -1724,6 +1724,42 @@ impl PyDiGraph {
         Err(NoSuitableNeighbors::new_err("No suitable neighbor"))
     }
 
+    /// Find a target node with a specific edge
+    ///
+    /// This method is used to find a target node that is an adjacent predecessor
```
Comment (mtreinish) [inline src/digraph.rs:1729]:
```suggestion
    /// This method is used to find an adjacent predecessor
```

**Review by mtreinish — COMMENTED** [review]
Overall I think this looks good, thanks for writing this. The only thing I think missing (besides the two nits on the docstring inline) is a negative test that tests we raise the appropriate error if there is no matching inbound edge found

**Comment by ewinston** [comment]
@mtreinish Thanks for the tip on how to implement this.

**Review by mtreinish — COMMENTED** [review]
The code LGTM now thanks. I just left inline comments to remove "adjacent" from the function name. I think it's a bit redundant with predecessor in the name already

The other thing I think we should have is a reno release note documenting the new feature.

**Review by mtreinish — APPROVED** [review]

**Pull request merged**